odoo/addons/base/models/ir_attachment.py

```diff
@@ -442,6 +442,19 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
 
         # sort result according to the original sort ordering
         result = [id for id in orig_ids if id in ids]
+
+        # If the original search reached the limit, it is important the
+        # filtered record set does so too. When a JS view recieve a
+        # record set whose length is bellow the limit, it thinks it
+        # reached the last page.
+        if len(orig_ids) == limit and len(result) < len(orig_ids):
+            result.extend(self._search(args,
+                offset=offset + len(orig_ids),
+                limit=limit - len(result),
```

Comment:
I think the best way to do this would be to keep `limit=limit`, then to limit the value returned by the recursive call to `_search`, i.e. `result.extend(self._search(...)[:limit - len(result)])`.

You will probably fetch way more than you need, but at least it's very unlikely that you'll have to query the database more than once